cron/worker/main.go

```diff
@@ -79,11 +85,18 @@ func processRequest(ctx context.Context,
 			return fmt.Errorf("error during result.AsJSON: %w", err)
 		}
 	}
-
 	if err := data.WriteToBlobStore(ctx, bucketURL, filename, buffer.Bytes()); err != nil {
 		return fmt.Errorf("error during WriteToBlobStore: %w", err)
 	}
 	log.Printf("Write to shard file successful: %s", filename)
+
+	pprof.StopCPUProfile()
+	cpuProfFilename := data.GetBlobFilename(
+		fmt.Sprintf("pprof.shard-%05d.cpu", batchRequest.GetShardNum()),
+		batchRequest.GetJobTime().AsTime())
+	if err := data.WriteToBlobStore(ctx, bucketURL, cpuProfFilename, cpuProf.Bytes()); err != nil {
+		return fmt.Errorf("error during WriteToBlobStore: %w", err)
```

Comment:
sometime in future, good to think about these error message, e.g. WriteToBlobStore is called in multiple places, so need to add more specific content in them, that it is cpu profiler specific etc.

Reply:
Acknowledged. Overall, the way we bubble up errors and log them needs to be improved.